### PR TITLE
Test/highlight correct nb

### PR DESCRIPTION
**Pull request by MaddyDev**
Tests for revealActiveDocumentInViewlet

### CODE REVIEW

**Comment by chlafreniere** [comment]
I'm curious why some files are showing lower test coverage here in related areas. Can you take a look, @MaddyDev?

**Comment by MaddyDev** [comment]
> I'm curious why some files are showing lower test coverage here in related areas. Can you take a look, @MaddyDev?

I was confused about that as well. Let me take a look.

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		}
 	}
 
-	async revealActiveDocumentInViewlet(uri?: vscode.Uri, shouldReveal: boolean = true): Promise<void> {
+	async revealActiveDocumentInViewlet(uri?: vscode.Uri, shouldReveal: boolean = true): Promise<BookTreeItem> | undefined {
```
Comment (corivera) [inline extensions/notebook/src/book/bookTreeView.ts:233]: Should the undefined part be in the angle brackets after BookTreeItem?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 2]
```diff
@@ -540,4 +541,8 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		let selectedBook = this.books.find(b => itemPath.toLowerCase().indexOf(b.bookPath.toLowerCase()) > -1);
 		return selectedBook;
 	}
+
+	dispose() {
+		this.dispose();
```
Comment (corivera) [inline extensions/notebook/src/book/bookTreeView.ts:546]: Is there a risk of stack overflows here?

File: extensions/notebook/src/extension.ts [hunk 3]
```diff
@@ -13,13 +13,11 @@ import { AppContext } from './common/appContext';
...
 import { NotebookUriHandler } from './protocol/notebookUriHandler';
-import { BookTreeViewProvider } from './book/bookTreeView';
-import { NavigationProviders, BuiltInCommands, unsavedBooksContextKey } from './common/constants';
+import { BuiltInCommands, unsavedBooksContextKey } from './common/constants';
 
 const localize = nls.loadMessageBundle();
 
-const BOOKS_VIEWID = 'bookTreeView';
-const PROVIDED_BOOKS_VIEWID = 'providedBooksView';
+
```
Comment (corivera) [inline extensions/notebook/src/extension.ts:20]: extra space here

File: extensions/notebook/src/book/bookTreeView.ts (hunk 2 above)
Comment (corivera) [reply extensions/notebook/src/book/bookTreeView.ts:546]: Actually, why do we need this method here? There's no parent class to call a dispose for, and we aren't doing anything in this method.

Comment (MaddyDev) [reply extensions/notebook/src/book/bookTreeView.ts]: I was playing around with exposing the extension level objects and added this for my testing, forgot to remove it.

File: extensions/notebook/src/test/book/book.test.ts [hunk 4]
```diff
@@ -28,23 +31,24 @@ export interface IExpectedBookItem {
 	nextUri?: string | undefined;
 }
 
-export function equalBookItems(book: BookTreeItem, expectedBook: IExpectedBookItem): void {
-	should(book.title).equal(expectedBook.title);
+export function equalBookItems(book: BookTreeItem, expectedBook: IExpectedBookItem, errorMsg?: string): void {
+	should(book.title).equal(expectedBook.title, 'Book titles do not match, expected ' + expectedBook.title + ' and got ' + book.title);
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:35]: Using interned strings is a lot easier to read

`` `Book titles do not match, espected ${expectedBook.title} and got ${book.title}` ``

File: extensions/notebook/src/test/book/book.test.ts [hunk 5]
```diff
@@ -114,6 +118,15 @@ describe('BookTreeViewProviderTests', function () {
 			await fs.writeFile(markdownFile, '');
 		});
 
+		it('bookProviders should be initialized on extension activate', async () => {
+			let notebookExtension: vscode.Extension<any> = vscode.extensions.getExtension('Microsoft.notebook');
+			await notebookExtension.activate();
+			appContext = notebookExtension.exports.getAppContext();
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:124]: .activate() returns the API directly, you don't need to go through exports.

`const appContext = (await vscode.extensions.getExtension('Microsoft.notebook').activate()).getAppContext()`

File: extensions/notebook/src/test/book/book.test.ts [hunk 6]
```diff
@@ -218,15 +224,65 @@ describe('BookTreeViewProviderTests', function () {
...
+				let revealActiveDocumentInViewletSpy = sinon.spy(bookTreeViewProvider, 'revealActiveDocumentInViewlet');
+				await azdata.nb.showNotebookDocument(notebookUri);
+				should(azdata.nb.notebookDocuments.find(doc => doc.fileName === notebookUri.fsPath)).not.be.undefined();
+				should(revealActiveDocumentInViewletSpy.calledOnce).be.true('revealActiveDocumentInViewlet should have been called');
+
+				notebookUri = vscode.Uri.file(notebook1Path);
+				await azdata.nb.showNotebookDocument(notebookUri);
+				should(azdata.nb.notebookDocuments.find(doc => doc.fileName === notebookUri.fsPath)).not.be.undefined();
+				should(revealActiveDocumentInViewletSpy.calledTwice).be.true('revealActiveDocumentInViewlet should have been called twice');
+				sinon.restore();
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:256]: Just use an afterEach hook at the top level describe to run sinon.restore after each test

File: extensions/notebook/src/test/book/book.test.ts [hunk 7]
```diff
@@ -338,13 +419,13 @@ describe('BookTreeViewProviderTests', function () {
 
 		it('should show error message if config.yml file not found', async () => {
 			await bookTreeViewProvider.currentBook.readBooks();
-			should(bookTreeViewProvider.currentBook.errorMessage.toLocaleLowerCase()).equal(('Failed to read book '+ bookTreeViewProvider.currentBook.bookPath +': ENOENT: no such file or directory, open \'' + configFile + '\'').toLocaleLowerCase());
+			should(bookTreeViewProvider.currentBook.errorMessage.toLocaleLowerCase()).equal(('Failed to read book ' + bookTreeViewProvider.currentBook.bookPath + ': ENOENT: no such file or directory, open \'' + configFile + '\'').toLocaleLowerCase());
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:422]: Why are you lower casing this? And just use the already defined constant - saves having to update this if we decide to modify the text later https://github.com/microsoft/azuredatastudio/blob/main/extensions/notebook/src/common/localizedConstants.ts#L36

File: extensions/notebook/src/test/book/book.test.ts [hunk 8]
```diff
@@ -461,14 +535,59 @@ describe('BookTreeViewProviderTests', function () {
...
+			sinon.stub(vscode.window, 'showOpenDialog').returns(Promise.resolve([vscode.Uri.file(rootFolderPath)]));
+			let openBookSpy = sinon.spy(bookTreeViewProvider, 'openBook');
+			await bookTreeViewProvider.openNewBook();
+
+			should(openBookSpy.calledOnce).be.true('openNewBook should have called openBook');
+			sinon.restore();
+		});
+
+		it('searchJupyterBooks should call command that opens Search view', async () => {
+			let executeCOmmandSpy = sinon.spy(vscode.commands, 'executeCommand');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:567]: extra capital o

File: extensions/notebook/src/test/book/book.test.ts (hunk 6 above)
Comment (MaddyDev) [reply extensions/notebook/src/test/book/book.test.ts:256]: I've used sinon in few tests only, so didn't make that afterEach hook, I guess if it isn't a lot of overhead I can add that.

File: extensions/notebook/src/test/book/book.test.ts [hunk 9]
```diff
@@ -461,14 +544,55 @@ describe('BookTreeViewProviderTests', function () {
...
+			let executeCommandSpy = sinon.spy(vscode.commands, 'executeCommand');
+			await bookTreeViewProvider.searchJupyterBooks(bookTreeViewProvider.books[0].bookItems[0]);
+			should(executeCommandSpy.calledWith('workbench.action.findInFiles')).be.true('searchJupyterBooks should have called command to open Search view');
+		});
+
+		it('saveJupyterBooks should prompt location and openBook', async () => {
+			let saveFolderPath = path.join(os.tmpdir(), `Book_${uuid.v4()}`);
+			await fs.mkdir(saveFolderPath);
+			sinon.stub(vscode.window, 'showOpenDialog').returns(Promise.resolve([vscode.Uri.file(saveFolderPath)]));
+			let executeCOmmandSpy = sinon.spy(vscode.commands, 'executeCommand');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:583]: Extra capital o here

**Pull request merged**